### ISSUE TITLE
Please force-yank rails-assets-angular

### ISSUE THREAD

**Issue by sheerun** [rubygems/rubygems.org#857]
Hello,

I'm representing https://rails-assets.org service. Someone used bundler's [CVE-2013-0334](http://bundler.io/blog/2014/08/14/bundler-may-install-gems-from-a-different-source-than-expected-cve-2013-0334.html) to perform attack on our service and created [rails-assets-angular](https://rubygems.org/gems/rails-assets-angular) gem with the same name as on our service.

Now every time someone `bundle update`, the rubygems gem is installed instead of ours..

Rubygems gem contains `alert('I'm in your browser')`. It affects our users. More details: https://github.com/rubygems/rubygems/issues/1148

I already asked author to remove this gem few days ago, he doesn't respond.

1. Could you force-yank this gem? Maybe delete it at all?
2. Can we maybe reserve `rails-assets-` prefix?

**Comment by evanphx** [comment]
@indirect Can you weigh in here, from the Bundler perspective? If the only reason the gem exists is to exploit this Bundler CVE, then I could see grounds to remove it.

As for reserving a prefix, that's not something we do, sorry.

**Comment by indirect** [comment]
@evan yeah, it looks like this gem exists solely to exploit the CVE. It's just a clone of the other gem with a JS alert added. 

@sheerun if you want to own gem names on rubygems.org you should handle that yourself.

**Comment by sheerun** [comment]
Are you suggesting to add 25 000 auto-generated gems to rubygems.org? ;)

**Comment by sheerun** [comment]
Any progress with yanking this gem? It's quite a chore.

**Comment by evanphx** [comment]
@sheerun  Did you hear back from the author?

**Comment by sheerun** [comment]
He doesn't respond or provided fake e-mail.

**Comment by evanphx** [comment]
@sheerun Sorry for the delay. After discussion we're going to attempt to contact the author one final time. They have until Friday to respond. If they don't respond or their justification for the gem is not valid, we'll yank the versions.

**Comment by imorcott** [comment]
It is Sunday evening...  ;)
I deleted my local ruby 2.1.5 the did a rvm install 2.1.5 and ruby p273 (2014-11-13 revision 48405) [x86_64-darwin14.0] was installed.
 Bundle version is 1.7.6
My Gemfile has:
 gem 'rails-assets-angular'

Then did a bundle update on my app and a bundle install.  And the problem persists.
The gem version for me that is producing a message 'I'm in your app' is:     
rails-assets-angular (9000.0.0)

So it looks like bundle 1.7.6 does not yet fully protect against this malicious gem and it appears that they have updated their version number too.

**Comment by TimMoore** [comment]
@imorcott to fully fix this:

1. You need to edit your `Gemfile` so that `rails-assets-angular` and any other rails-assets gems are contained within a source block:
    ```
    source 'https://rails-assets.org' do
      gem 'rails-assets-angular'
    end
    ```
2. Remove the top level `source 'https://rails-assets.org'` and only use it with this block syntax.
3. `gem uninstall rails-assets-angular` to make sure the broken version is removed

I'd also recommend updating to the latest version of bundler (1.7.13) with `gem install bundler`. There have been some bug fixes since 1.7.6.

When you use the block syntax, it won't use the version on rubygems.org and so you won't be held up on it being removed.

Even if the broken version of the gem is removed from the server, if you already have it installed it will keep using the installed version, so please be sure to remove it completely.

**Comment by indirect** [comment]
@sheerun the gem has been yanked, and I have temporarily taken over the gem so no new versions can be pushed. In general, please keep in mind that anyone can push a gem to rubygems.org with any name that has not been taken yet!
